src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -25,6 +25,31 @@ public abstract class TrainCatalogBase
         [BestFriend]
         internal IHostEnvironment Environment => Host;
 
+        /// <summary>
+        /// A pair of datasets, for the train and test set.
+        /// </summary>
+        public struct TrainTestData
+        {
```

Comment:
I could see us wanting to expand this in the future to also include a validation set, so it could be prudent to keep the name a bit vague. `PartitionedData`?

Reply:
What would be summary for PartitionedData?
This one used as output for TrainTestSplit function, if we ever introduce TrainTestValidationSplit I would prefer to create another object for that.

---
In reply to: [255725077](https://github.com/dotnet/machinelearning/pull/2507#discussion_r255725077) [](ancestors = 255725077)

Reply:
Good point!

---
In reply to: [255725959](https://github.com/dotnet/machinelearning/pull/2507#discussion_r255725959) [](ancestors = 255725959,255725077)